### ISSUE TITLE
Build failure on 18.10 cosmic (gtest)

### ISSUE THREAD

**Issue by Saviq** [canonical/multipass#458]
```
/usr/lib/ccache/c++  -DCARES_STATICLIB -DFMT_HEADER_ONLY=1 -DMULTIPASS_PLATFORM_LINUX -DQT_CORE_LIB -DQT_NETWORK_LIB -DQT_NO_DEBUG -I../include -Igen -I../ -I../3rd-party/grpc/third_party/googletest/googletest -I../3rd-party/grpc/third_party/googletest/googletest/include -I../3rd-party/grpc/third_party/googletest/googlemock -I../3rd-party/grpc/third_party/googletest/googlemock/include -I../3rd-party/fmt/include -I../3rd-party/libssh/libssh/include -I../3rd-party/grpc/include -I../3rd-party/grpc/third_party/protobuf/src -I3rd-party/grpc/third_party/cares/cares -I../3rd-party/grpc/third_party/cares/cares -I../3rd-party/yaml-cpp/include -I../3rd-party/xz-decoder/xz-embedded/linux/include/linux -I../3rd-party/premock -isystem /usr/include/x86_64-linux-gnu/qt5 -isystem /usr/include/x86_64-linux-gnu/qt5/QtCore -isystem /usr/lib/x86_64-linux-gnu/qt5/mkspecs/linux-g++ -isystem /usr/include/x86_64-linux-gnu/qt5/QtNetwork -Werror -Wall -pedantic -Wextra -Wno-unused-parameter -Wempty-body -Wformat-security -Winit-self -Warray-bounds -fPIC -Wcast-align -Wno-expansion-to-defined -std=c++14 -MD -MT tests/CMakeFiles/multipass_tests.dir/test_sftpserver.cpp.o -MF tests/CMakeFiles/multipass_tests.dir/test_sftpserver.cpp.o.d -o tests/CMakeFiles/multipass_tests.dir/test_sftpserver.cpp.o -c ../tests/test_sftpserver.cpp                                                                                                            
In file included from ../3rd-party/grpc/third_party/googletest/googletest/include/gtest/gtest-param-test.h:192,
                 from ../3rd-party/grpc/third_party/googletest/googletest/include/gtest/gtest.h:62,
                 from ../tests/sftp_server_test_fixture.h:24,
                 from ../tests/test_sftpserver.cpp:18:
../3rd-party/grpc/third_party/googletest/googletest/include/gtest/internal/gtest-param-util.h: In instantiation of ‘class testing::internal::ParameterizedTestFactory<WhenInvalidMessageReceived_replies_failure_Test>’:                     
../3rd-party/grpc/third_party/googletest/googletest/include/gtest/internal/gtest-param-util.h:453:12:   required from ‘testing::internal::TestFactoryBase* testing::internal::TestMetaFactory<TestCase>::CreateTestFactory(testing::internal::TestMetaFactory<TestCase>::ParamType) [with TestCase = WhenInvalidMessageReceived_replies_failure_Test; testing::internal::TestMetaFactory<TestCase>::ParamType = {anonymous}::MessageAndReply]’                                             
../3rd-party/grpc/third_party/googletest/googletest/include/gtest/internal/gtest-param-util.h:452:28:   required from here                                                                                                                   
../3rd-party/grpc/third_party/googletest/googletest/include/gtest/internal/gtest-param-util.h:408:7: error: ‘testing::internal::ParameterizedTestFactory<WhenInvalidMessageReceived_replies_failure_Test>’ has a field ‘testing::internal::ParameterizedTestFactory<WhenInvalidMessageReceived_replies_failure_Test>::parameter_’ whose type uses the anonymous namespace [-Werror=subobject-linkage]                                                                                      
 class ParameterizedTestFactory : public TestFactoryBase {
       ^~~~~~~~~~~~~~~~~~~~~~~~
cc1plus: all warnings being treated as errors
```

**Comment by Saviq** [comment]
Fixed by: #624

**Comment by townsend2010** [comment]
Closing via #624.
